system/Router/RouteCollection.php

```diff
@@ -1565,7 +1565,20 @@ private function checkHostname($hostname): bool
             return false;
         }
 
-        return strtolower($this->httpHost) === strtolower($hostname);
+        // Hostname has multiple hostname
+        if(is_array($hostname)) 
+        {
+            $hasMatch = false;
+            foreach ($hostname as $host) {
+                if (strtolower($this->httpHost) === strtolower($host)) {
+                    $hasMatch = true;
+                }
+            }
+            return $hasMatch;
```

Comment:
```suggestion
            $hostnameLower = array_map('strtolower', $hostname);

            return in_array(strtolower($this->httpHost), $hostnameLower, true);
```

No need looping, by the way needed unit testing for this